style(lexer): reformat `Kind` matchers

overlookmotel opened the pull request:
Pure style refactor:

* Use `==` instead of `matches!` when matching a single variant.
* Reformat multi-line `matches!` blocks.

(found this old branch lying around from March which I'd never got around to submitting)

Boshen commented:
### Merge activity

* **Jul 25, 4:10 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
style(lexer): reformat `Kind` matchers (#12520)

Pure style refactor:

* Use `==` instead of `matches!` when matching a single variant.
* Reformat multi-line `matches!` blocks.

(found this old branch lying around from March which I'd never got around to submitting)
